### DISCUSSION TITLE
trained model (train_unet_demo) gives bad images.h5

### DISCUSSION THREAD

**Discussion by marinapapageorgiou** [facebookresearch/fastMRI#350] [Q&A]
Hello everyone,
I am studying singlecoil (Unet model) and knee.

I run the pretrained model (run_pretrained_unet_inference.py) and the results are these images:
![1](https://github.com/user-attachments/assets/a59f6108-5fa1-4564-8fa7-f6d16240bdb0)
![2](https://github.com/user-attachments/assets/a5af5a25-6f13-496e-9fab-da169100726c)

After I trained the Unet model (train_unet_demo.py) and I tested it. The images of the testing are these:
![3](https://github.com/user-attachments/assets/d54bd59f-dde7-4d2d-baa5-980fd35483ea)
![4](https://github.com/user-attachments/assets/151ac2e7-984b-431a-9f79-e3b98f1ea4bd)

If it is useful I run the model with max_epochs=50 and lr=0.001, # learning rate. And i tried also with  max_epochs=50 and lr=0.0003, # learning rate. But in both cases the best epoch that is keeps as the best is epoch=41.
Also, I am using only these files for dataset
knee_singlecoil_train
knee_singlecoil_val
knee_singlecoil_test

So my question is, What is wrong with the testing images of my training model? Is the dataset not big enough? Is the learning rate not appropriate? Is something else? Does anyone know?

**Comment by mmuckley** [comment]
Converting to a discussion as this does not seem to document a bug with the software.

@marinapapageorgiou have you trained your model on the fastMRI single-coil data and looked at those images vs. the published results? If you're training on your own data, it's more of a research question and I don't think I can help without further details.

**Reply by marinapapageorgiou** [reply]
Hello @mmuckley,

I did not use my own data. I used the dataset given by the article (knee_singlecoil_train, knee_singlecoil_val, knee_singlecoil_test). Is the datset sufficient? Or am I doing something else wrong?

Also, the dataset folder (knee_singlecoil_train) contains 973 files. Is this the correct dataset for training the model?

**Reply by mmuckley** [reply]
@marinapapageorgiou I think that is the right dataset. What is the undersampling ratio?

There is definitely something wrong with the results. The exact code with no modifications that I used to generate the pretrained model is here: https://github.com/facebookresearch/fastMRI/blob/main/fastmri_examples/unet/unet_reproduce_20201111.py/unet_knee_sc_leaderboard.py.

As you can see, it was run way back in November of 2020, so there may be some changes to dependencies since then that are shifting things.

**Reply by marinapapageorgiou** [reply]
> @marinapapageorgiou I think that is the right dataset. What is the undersampling ratio?
> 
> There is definitely something wrong with the results. The exact code with no modifications that I used to generate the pretrained model is here: https://github.com/facebookresearch/fastMRI/blob/main/fastmri_examples/unet/unet_reproduce_20201111.py/unet_knee_sc_leaderboard.py.
> 
> As you can see, it was run way back in November of 2020, so there may be some changes to dependencies since then that are shifting things.

@mmuckley, I changed the parameter chans=32 instead of 256 and the versions of the environment and everything worked good.
